src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticQueryService.cs

```diff
@@ -33,6 +33,8 @@ public async Task<ElasticTopDocs> SearchAsync(string indexName, string query)
             if (_elasticClient == null)
```

Comment:
If you check line 41 and after, they are using `_elasticClient`, which is possible for NRE. The condition in line 33 is useless because it doesn't returns if the field is null

Reply:
Ok, so something has changed that make this fail. It used to work.

Reply:
Please have a look quickly at the service and you will know what I mean.

> so something has changed that make this fail

I can have a look to the history logs

Reply:
Well it's been a year a worked on that one and I need to remember details. But here we are supposed to define a default analyzer and we added the userDefined one afterward.

Reply:
@Skrypt it's there since your PR #11052

Reply:
This is probably returning an NRE because of the way we are registering the services now.

Reply:
I forgot to cleanup probably 😄

Reply:
We need to find a new way to do this then. The goal was to display a message when the service is not properly setup but they changed the startup.cs code so we can't do this anymore I guess. Maybe there is an alternate way.

Reply:
I would simply remove the entire If code block for now.

Reply:
> We need to find a new way to do this then. The goal was to display a message when the service is not properly setup but they changed the startup.cs code so we can't do this anymore I guess. Maybe there is an alternate way.

If that's your intention, we might return null, then check in the controller / driver for nullability and display a notification